src/protocols/dhcpv4/encode.c

```diff
@@ -235,20 +265,25 @@ static ssize_t encode_rfc_hdr(fr_dbuff_t *dbuff,
 		FR_PROTO_TRACE("Encoded value is %zu byte(s)", len);
 		FR_PROTO_HEX_DUMP(dbuff->p, fr_dbuff_used(&work_dbuff), NULL);
 
-		if ((hdr[1] + len) <= 255) {
-			hdr[1] += len;
-			FR_PROTO_TRACE("%u byte(s) available in option", 255 - hdr[1]);
+		if ((option_len + len) <= 255) {
+			option_len += len;
+			fr_dbuff_advance(&hdr, 1);
+			fr_dbuff_in(&hdr, option_len);
+			fr_dbuff_advance(&hdr, -2);
```

Comment:
Hm I know it works but it's weird behaviour.  Maybe add a `fr_dbuff_retreat` as that's the atonym...

Reply:
Usually you'd use markers for this kind of thing though because you can be sure the data hasn't been shifted out.  Consider an extension function which writes out data to a socket and shifts it to the left when the buffer is full.

Reply:
Remember you can use fr_sbuff_set on markers now....

Reply:
Good point. I could add another marker so hdr can stay as is. It would only be used for the length, so it needn't be retreated..

Reply:
Pushed change to leave hdr alone-no more retreating. I did rename markers, but it seemed worth it for consistency and indicating the markers' purpose.